### PR TITLE
Handle close of listeners to QC Auto Vetter to trigger removal of graph listneer if no listeners to auto vetter

### PR DESCRIPTION
**Pull request by serpens24**
### Prerequisites

- [X] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Added code to QualityControlAutoVetter.java to remove graph change listner if no listeners remain to QualityControlAutoVetter.
Updated DataAccessViewTopComponent.java and DefaultQualityControlAutoButton.java to modify when listeners to QualityControlAutoVetter are added/removed to ensure closing the view removes all listeners.
QualityControlView already correctly removes listeners when closed.

### Alternate Designs

nil

### Why Should This Be In Core?

Improve performance by removing redundant calls to QualityControlView.graphChanged.

### Benefits

As above

### Possible Drawbacks

Nil known

### Verification Process

To test:
* in debugger, add following breakpoints to QualityControlAutoVetter.java:
	1) line 147 in first line of graphChanged
    2) line 270 in addListener where addGraphChangeListener is called
	3) line 298 in removeListener where removeGraphChangeListener is called	
* start application
* open QCV and confirm no breakpoints are hit
* open a new graph and confirm breakpoint 1) is hit
* close QCV and confirm breakpoint 3) is hit
* put graph in draw mode - confirm no breakpoints are hit
* open QCV and confirm breakpoints 1) and 2) are hit
* draw a new node and confirm breakpoint 1) is hit
* open DAV and confirm no breakpoints are hit
* close QCV and confirm breakpoint 1) is hit
* draw a new node and confirm breakpoint 1) is hit
* close DAV and confirm breakpoint 3) is hit
* draw a new node and confirm breakpoint 1) is not hit
* open DAV and confirm breakpoint 2 is hit
* draw a new node and confirm breakpoint 1) is hit
* open QCV and confirm breakpoint 1) is hit
* close DAV and confirm breakpoint 1) is hit
* close QCV and confirm breakpoint 3) is hit

### Applicable Issues

#916

### CODE REVIEW

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java [hunk 1]
```diff
@@ -144,19 +155,49 @@ protected void componentShowing() {
...
+
+    /**
+     * Add or remove all quality control auto vetter listeners. These listeners
+     * are tied to DefaultQualityControlAutoButton buttons found nested within
+     * the button toolbar.
+     *
+     * @param add Should quality control auto vetter listeners be added (true)
+     * or removed (false).
+     */
+    private void ManageQualityControlListeners(boolean add) {
```
Comment (OrionsGuardian) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:174]: The method/function names should begin with a lower case character.
Rename this to: manageQualityControlListeners

File: CoreQualityControlView/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButton.java [hunk 2]
```diff
@@ -105,4 +100,22 @@ public DefaultQualityControlAutoButton copy() {
         return new DefaultQualityControlAutoButton();
     }
 
+    /**
+     * Add this button as a listener to the quality control auto vetter. This
+     * allows the containing top component to effectively subscribe/unsubscribe
+     * listeners as the components are opened or closed.
+     */
+    public void AddQCListener() {
```
Comment (OrionsGuardian) [inline CoreQualityControlView/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButton.java:108]: Rename this to: addQCListener

File: CoreQualityControlView/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButton.java [hunk 3]
```diff
@@ -105,4 +100,22 @@ public DefaultQualityControlAutoButton copy() {
...
+        QualityControlAutoVetter.getInstance().addListener(this);
+        QualityControlAutoVetter.getInstance().invokeListener(this);
+    }
+
+    /**
+     * Remove this button as a listener from the quality control auto vetter.
+     * This allows the containing top component to effectively
+     * subscribe/unsubscribe listeners as the components are opened or closed.
+     */
+    public void RemoveQCListener() {
```
Comment (OrionsGuardian) [inline CoreQualityControlView/src/au/gov/asd/tac/constellation/views/qualitycontrol/widget/DefaultQualityControlAutoButton.java:118]: Rename this to: removeQCListener

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java [hunk 4]
```diff
@@ -144,19 +155,49 @@ protected void componentShowing() {
 
     @Override
     protected void handleNewGraph(final Graph graph) {
-        System.setProperty("dav.graph.ready", Boolean.FALSE.toString());
```
Comment (OrionsGuardian) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:147]: Why has this code been removed?

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java [hunk 5]
```diff
@@ -144,19 +155,49 @@ protected void componentShowing() {
...
             getDataAccessPane().update(graph);
-            Platform.runLater(() ->
+            Platform.runLater(() -> 
                 DataAccessUtilities.loadDataAccessState(getDataAccessPane(), graph)
             );
         }
-        Platform.runLater(() ->
-                // nested runLater so it runs after all the other triggered processes for opening a graph have been run
-                Platform.runLater(() -> System.setProperty("dav.graph.ready", Boolean.TRUE.toString()))
-        );
```
Comment (OrionsGuardian) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:157]: Why has this code been removed?

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java (hunk 1 above)
Comment (serpens24) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:174]: Done

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java (hunk 4 above)
Comment (serpens24) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:147]: Done. This got missed in the merge

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java (hunk 5 above)
Comment (serpens24) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:157]: Restored. This got missed in the merge

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java [hunk 6]
```diff
@@ -156,7 +166,42 @@ protected void handleNewGraph(final Graph graph) {
...
+
+    /**
+     * Add or remove all quality control auto vetter listeners. These listeners
+     * are tied to DefaultQualityControlAutoButton buttons found nested within
+     * the button toolbar.
+     *
+     * @param add Should quality control auto vetter listeners be added (true)
+     * or removed (false).
+     */
+    private void manageQualityControlListeners(boolean add) {
```
Comment (OrionsGuardian) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/DataAccessViewTopComponent.java:178]: final

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]

**Review by OrionsGuardian — APPROVED** [review]

**Pull request merged**